src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs

```diff
@@ -323,7 +323,7 @@ private void FormatDocumentCreatedFromTemplate(IVsHierarchy hierarchy, uint item
 
             // Organize using directives
             addedDocument = ThreadHelper.JoinableTaskFactory.Run(() => OrganizeUsingsCreatedFromTemplateAsync(addedDocument, cancellationToken));
-            rootToFormat = addedDocument.GetSyntaxRootSynchronously(cancellationToken);
+            rootToFormat = ThreadHelper.JoinableTaskFactory.Run(() => addedDocument.GetRequiredSyntaxRootAsync(cancellationToken));
```

Comment:
Would it be better to combine this with the call above to reduce the thread thrashing?

Reply:
This code runs only once when a new document is created from template. Combining wouldn't really make a perf difference here.